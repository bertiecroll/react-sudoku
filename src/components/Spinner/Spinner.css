.Spinner {
  margin-top: 3rem;
  text-align: center;
  color: #777;
  display: flex;
  flex-direction: column;
}

.Spinner-logo {
  animation: App-logo-spin infinite 2s ease-in-out;
  height: 180px;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
