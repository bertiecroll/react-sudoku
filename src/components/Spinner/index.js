export { Spinner } from './Spinner'
export { default } from './Spinner'
