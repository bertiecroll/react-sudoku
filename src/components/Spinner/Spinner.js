import React from 'react'
import './Spinner.css'
import logo from 'images/logo.svg';

export default function Spinner () {
  return (
    <div className="Spinner">
      <img src={logo} className="Spinner-logo" alt="logo" />
      <h4>Loading Puzzle...</h4>
    </div>
  )
}
