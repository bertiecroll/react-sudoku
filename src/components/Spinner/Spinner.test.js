import React from 'react'
import { shallow } from 'enzyme'
import Spinner from 'components/Spinner'

describe('Spinner', () => {
  it('renders logo with loading text', () => {
    const wrapper = shallow(<Spinner />)
    expect(wrapper).toMatchSnapshot()
  })
})
