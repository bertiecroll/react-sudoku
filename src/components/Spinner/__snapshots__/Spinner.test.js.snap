// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner renders logo with loading text 1`] = `
<div
  className="Spinner"
>
  <img
    alt="logo"
    className="Spinner-logo"
    src="logo.svg"
  />
  <h4>
    Loading Puzzle...
  </h4>
</div>
`;
